src/components/quote.tsx

```diff
@@ -1,16 +1,58 @@
 import Component from "inferno-component";
 import * as moment from "moment";
 
+import classifyQuote from "../lib/classifyquote";
+
 require("../../style/quote.scss");
 
-const Quote = ({id, author, body, addedAt}) =>
-    <article class="quote">
-        <section class="quote-body">
-            <pre class="quote-text">
-                {JSON.stringify(body, null, 2)}
-            </pre>
+const QuoteSegment = ({speaker, displayIcon, body}) => {
+    let showUserInfo = speaker || displayIcon;
+    return (
+        <section class={showUserInfo ? "quote-segment" : "quote-segment follow"}>
+            {showUserInfo ?
+                [
+                    <img class="quote-segment-icon" src={displayIcon}/>,
+                    <em class="quote-segment-speaker">{speaker}</em>
+                ]
+                : null}
+            <p class="quote-segment-body">{body}</p>
         </section>
-        <aside class="quote-timestamp">added by {author} {moment(new Date(addedAt)).fromNow()}</aside>
-    </article>;
+        );
+};
+
+const Quote = ({id, author, body, addedAt}) => {
+    let bodyElements;
+    if (body.type === "unrecognized") {
+        bodyElements = body.message;
+    } else {
+        bodyElements = [];
+        let previousSpeaker = undefined;
+        for (let message of body.messages) {
+            if (previousSpeaker === message.speaker) {
+                bodyElements.push(
+                    <QuoteSegment
+                        body={message.body}/>
+                );
+            } else {
+                bodyElements.push(
+                    <QuoteSegment
+                        displayIcon={"https://placehold.it/64x64"}
+                        speaker={message.speaker}
+                        body={message.body}/>
+                );
+            }
+            previousSpeaker = message.speaker;
+        }
+    }
+
+    return (
```

Comment:
Threw some border color on that shizz

![image](https://cloud.githubusercontent.com/assets/1174858/25728104/08f3de92-30fb-11e7-872a-3dd4b2f42f1d.png)